mtags-shared/src/main/scala/scala/meta/internal/mtags/CoursierComplete.scala

```diff
@@ -20,35 +20,70 @@ class CoursierComplete(scalaVersion: String) {
       else scalaVersion.split('.').take(2).mkString(".")
     )
 
+  private def completions(s: String): List[String] = {
+    val futureCompletions = Future {
+      api.withInput(s).complete().getCompletions().asScala.toList
+    }
+    try Await.result(futureCompletions, 10.seconds)
+    catch {
+      case _: Throwable => Nil
+    }
+  }
+
   def complete(
       dependency: String,
-      includeScala: Boolean = true
+      includeScala: Boolean = true,
+      supportNonJvm: Boolean = true
   ): List[String] = {
+    // Version completions
+    if (dependency.replaceAll(":+", ":").count(_ == ':') == 2) {
+      versionCompletions(dependency, supportNonJvm)
+    } else {
+      val javaCompletions = completions(dependency)
+      val scalaCompletions =
+        if (
+          includeScala &&
+          dependency.endsWith(":") && dependency.count(_ == ':') == 1
+        )
+          completions(dependency + ":").map(":" + _)
+        else List.empty
 
-    def completions(s: String): List[String] = {
-      val futureCompletions = Future {
-        api.withInput(s).complete().getCompletions().asScala.toList
-      }
-      try Await.result(futureCompletions, 10.seconds)
-      catch {
-        case _: Throwable => Nil
-      }
+      (scalaCompletions ++ javaCompletions).distinct
     }
+  }
 
-    val javaCompletions = completions(dependency)
-    val scalaCompletions =
-      if (
-        includeScala &&
-        dependency.endsWith(":") && dependency.count(_ == ':') == 1
-      )
-        completions(dependency + ":").map(":" + _)
-      else List.empty
+  private def versionCompletions(
+      dependency: String,
+      supportNonJvm: Boolean
+  ): List[String] = {
+    val (adjusted, hasDoubleColon) = adjustDoubleColon(dependency)
+    val sortedCompletions = completions(adjusted).sortWith(
+      Version.fromString(_) >= Version.fromString(_)
+    )
+    if (!hasDoubleColon && supportNonJvm)
+      sortedCompletions.map(":" + _)
+    else sortedCompletions
+
+  }
 
-    val allCompletions = (scalaCompletions ++ javaCompletions).distinct
-    // Attempt to sort versions in reverse order
-    if (dependency.replaceAll(":+", ":").count(_ == ':') == 2)
-      allCompletions.sortWith(Version.fromString(_) >= Version.fromString(_))
-    else allCompletions
+  private def adjustDoubleColon(dependency: String): (String, Boolean) = {
```

Comment:
Could you add a comment, why it's needed? Is it because coursier doesn't support `::` at the end?

Could you raise an issue in coursier also?

Reply:
Also, maybe:
```
def adjustDoubleColon(dependency: String): (String, Boolean) = {
  val doubleColon = dependency.lastIndexOf("::")
  val firstColon = dependency.indexOf(":")

  if (doubleColon > firstColon) {
    val depString = (dependency.substring(0, doubleColon) + ":" +
      dependency.substring(doubleColon + 2, dependency.length()))
    (depString, true)
  } else (dependency, false)

}
```

Reply:
This looks way better,  thanks